fips/usr/lib/systemd/system/snap-initramfs-mounts.service.d/fips-img.conf

```diff
@@ -0,0 +1,15 @@
+[Unit]
+After=systemd-modules-load.service
+
+# This is for kernels that are not booted from an EFI environment. This probably
+# means a FIT image on arm64.
+ConditionFirmware=!uefi
```

Comment:
Note that drops-ins (snippet in $UNIT.d/*.conf) apply for the $UNIT in question in full, not just to the unit itself.

```
$ systemctl cat hello
# /etc/systemd/system/hello.service
[Service]
Type=oneshot
ExecStart=echo hello

# /etc/systemd/system/hello.service.d/nonuefi.conf
[Unit]
ConditionFirmware=!uefi

[Service]
ExecStartPre=echo non-uefi


# /etc/systemd/system/hello.service.d/uefi.conf
[Unit]
ConditionFirmware=uefi

[Service]
ExecStartPre=echo uefi
```

Given the two drop-ins declare mutually exclusive conditions:

```
$ sudo systemctl status hello
○ hello.service
     Loaded: loaded (/etc/systemd/system/hello.service; static)
    Drop-In: /etc/systemd/system/hello.service.d
             └─nonuefi.conf, uefi.conf
     Active: inactive (dead)

Jul 24 22:24:35 xnox-Inspiron-7400 systemd[1]: Condition check resulted in hello.service being skipped.
```

It would result in snap-initramfs-mounts.service to never start, because condition failed, and will result in all systems to panic.

This seems to imply, that implementation was proposed, without design nor testing.

On this basis this is a reject, but also please learn how drop-ins works - because if such pattern is used elsewhere, it likely does not result in the behavior you desire.